main/settings.py

```diff
@@ -206,21 +206,21 @@
 }
 
 MIDDLEWARE = [
+    'middlewares.cache.UpdateCacheForUserMiddleware',
```

Comment:
Since we are using `django.contrib.auth.middleware.AuthenticationMiddleware` in this middlware. We will need to move this after that.